### PR TITLE
Support JNIFuse on macOS

### PR DESCRIPTION
**Pull request by apc999**
A few improvements:

1. Fix https://github.com/Alluxio/alluxio/issues/12720 to support using JNI fuse on MacOS
2. Fix umount issue on MacOS

### CODE REVIEW

File: integration/fuse/src/main/java/alluxio/jnifuse/struct/FileStat.java [hunk 1]
```diff
@@ -72,13 +72,13 @@ public static boolean S_ISLNK(int mode) {
   public FileStat(ByteBuffer buffer) {
     super(buffer);
     if (OSUtils.isMacOS()) {
-      st_dev = new Unsigned64();
-      st_mode = new Unsigned32();
-      st_nlink = new Unsigned64();
+      st_dev = new Unsigned32();
```
Comment (gpang) [inline integration/fuse/src/main/java/alluxio/jnifuse/struct/FileStat.java:75]: How do we know what the types are supposed to be?

**Review by gpang — COMMENTED** [review]
@apc999 Thanks! Just a minor question.

**Comment by gpang** [comment]
```Error: 6.427 [ERROR] Failures: 
Error: 6.427 [ERROR] alluxio.jnifuse.struct.FileStatTest.offset(alluxio.jnifuse.struct.FileStatTest)
Error: 6.427 [ERROR]   Run 1: FileStatTest.offset:32 expected:<16> but was:<12>
Error: 6.428 [ERROR]   Run 2: FileStatTest.offset:32 expected:<16> but was:<12>
```

File: integration/fuse/src/main/java/alluxio/jnifuse/struct/FileStat.java (hunk 1 above)
Comment (apc999) [reply integration/fuse/src/main/java/alluxio/jnifuse/struct/FileStat.java:75]: basically I checked out the src code of corresponding definition in MacOS and Linux in their kernels.
It looks like Linux and MacOS are using different data types for these structures.
We may need an indirection here.

**Comment by LuQQiu** [comment]
Failure in the tests
```
09:03:51.980 [INFO] Running alluxio.jnifuse.struct.FileStatTest
Error: 2.308 [ERROR] Tests run: 3, Failures: 2, Errors: 0, Skipped: 0, Time elapsed: 0.326 s <<< FAILURE! - in alluxio.jnifuse.struct.FileStatTest
Error: 2.309 [ERROR] offset(alluxio.jnifuse.struct.FileStatTest)  Time elapsed: 0.176 s  <<< FAILURE!
java.lang.AssertionError: expected:<16> but was:<12>
```
I commented on issue #12720 about the command testing results in macOS, PTAL, thanks!

**Comment by apc999** [comment]
@LuQQiu @maobaolong  this PR is ready for review

**Review by maobaolong — COMMENTED** [review]
Thanks for this feature. It look good overall, I left a couple of question inline, please take a look.

BTW, I test this can work on my env. `Macos Catalina 10.15.4`

**Review by LuQQiu — APPROVED** [review]
LGTM! Thanks for supporting JNI on macOS and solve the deployment painpoint!

File: integration/fuse/src/main/java/alluxio/jnifuse/struct/FileStat.java [hunk 2]
```diff
@@ -11,132 +11,30 @@
...
-  public static final int S_IXGRP = 0000010; // execute/search permission, group
-  public static final int S_IROTH = 0000004; // read permission, other
-  public static final int S_IWOTH = 0000002; // write permission, other
-  public static final int S_IXOTH = 0000001; // execute permission, other
+public class FileStat extends ru.serce.jnrfuse.struct.FileStat {
 
-  public static final int ALL_READ = S_IRUSR | S_IRGRP | S_IROTH;
-  public static final int ALL_WRITE = S_IWUSR | S_IWGRP | S_IWOTH;
-  public static final int S_IXUGO = S_IXUSR | S_IXGRP | S_IXOTH;
+  private final ByteBuffer buffer;
```
[the remark this answers is not in the document]
Comment (apc999) [reply integration/fuse/src/main/java/alluxio/jnifuse/struct/FileStat.java:21]: in dirFiller, we need to pass the handle of bytebuffer

File: integration/fuse/src/main/java/alluxio/jnifuse/struct/FuseContext.java [hunk 3]
```diff
@@ -11,17 +11,26 @@
...
 import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
 
 // TODO(iluoeli): Add more fields
-public class FuseContext extends Struct {
-
-  public final Unsigned32 uid = new Unsigned32();
-  public final Unsigned32 gid = new Unsigned32();
+public class FuseContext extends ru.serce.jnrfuse.struct.FuseContext {
+  private final ByteBuffer buffer;
```
[the remark this answers is not in the document]
Comment (apc999) [reply integration/fuse/src/main/java/alluxio/jnifuse/struct/FuseContext.java:21]: correct. Saw your other PR.

**Comment by apc999** [comment]
alluxio-bot, merge this please

**Pull request merged**